lib/purgatory/purgatory.rb

```diff
@@ -44,9 +44,14 @@ def soul_with_changes
   end
 
   def approve!(approver = nil)
-    return false if approved? 
-    if soul_with_changes.save
-      soul.send(performable_method[:method],*performable_method[:args]) if performable_method
+    return false if approved?
+    success = nil
```

Comment:
Hmm so is there a case where we'd make a Purgatory with no changes at all? Previously the code would always do `soul_with_changes.save` without even checking if there were requested_changes. So maybe we should do the same?

How about:

```
if performable_method.present?
  success = soul.public_send(performable_method[:method],*performable_method[:args])
else
  success = soul_with_changes.save
end

if success
  ...
```

Note that aside from the logic change I also am suggesting a couple other small tweaks:

1. I prefer to check `performable_method.present?`. That way we know this is an object that we're checking if it exists. The way it was before, the reader has no way to know if this is a check for object existence or a boolean.

2. We should have used `public_send` instead of `send`, since there shouldn't be a purgatory for private methods.

@thisduck @MarissaMac do you think it's ok to do #2? Theoretically we could break someone's system if they've been using purgatory for private methods. I guess we can just make a new overall version. Currently we're on 3.0.0 so we'd probably need to make it 4.0.0?